PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/action/RunConfigurationActionUtils.kt

```diff
@@ -104,9 +102,20 @@ object RunConfigurationActionUtils: ILogger {
                             "Continue Anyway",
                             Messages.getErrorIcon()) == YES
 
-    fun checkRunnerSettings(runProfile: AbstractRunConfiguration, runner: ProgramRunner<RunnerSettings>) {
+    private fun checkRunnerSettings(runProfile: AbstractRunConfiguration, runner: ProgramRunner<RunnerSettings>) {
         assertInDispatchThread()
 
         runProfile.checkRunnerSettings(runner, null, null)
     }
+
+    private fun checkAndPrepareRunProfileState(runProfile: AbstractRunConfiguration,
+                                               runner: ProgramRunner<RunnerSettings>,
+                                               ideaSchedulers: IdeaSchedulers): Observable<out Any?> {
+        if (runProfile !is RunProfileStatePrepare<*>) {
+            return empty()
```

Comment:
Throw a exception or add some logs here?

Reply:
It's expected. No need for that.